File: C# Examples/02 One Button PDF Publishing/SavePDF/SolidWorksMacro.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows;
using System.Windows.Forms;

using SolidWorks.Interop.sldworks;
using SolidWorks.Interop.swconst;


namespace SavePDF
{
    public partial class SolidWorksMacro
    {
        public void Main()
        {
            ModelDoc2 swDoc = null;
            int longstatus = 0;
            swDoc = ((ModelDoc2)(swApp.ActiveDoc));
            string FilePath = "";
            string NewFilePath = "";
            FilePath = swDoc.GetPathName();
            NewFilePath = System.IO.Path.ChangeExtension(FilePath, ".PDF");
            //
            // Save As
            longstatus = swDoc.SaveAs3(NewFilePath, 0, 0);
            MessageBox.Show("Saved " + NewFilePath, "Save message",
            MessageBoxButtons.OK, MessageBoxIcon.Information);
            return;
        }

        // The SldWorks swApp variable is pre-assigned for you.
        public SldWorks swApp;

    }
}

File: 02 One Button PDF Publishing/SavePDF/SolidWorksMacro.vb
Imports SolidWorks.Interop.sldworks
Imports SolidWorks.Interop.swconst
Imports System.Runtime.InteropServices
Imports System


Partial Class SolidWorksMacro
    Public Sub main()

        Dim swDoc As ModelDoc2 = Nothing
        Dim swPart As PartDoc = Nothing
        Dim swDrawing As DrawingDoc = Nothing
        Dim swAssembly As AssemblyDoc = Nothing
        Dim boolstatus As Boolean = False
        Dim longstatus As Integer = 0
        Dim longwarnings As Integer = 0
        swDoc = CType(swApp.ActiveDoc, ModelDoc2)
        swDoc.ClearSelection2(True)
        '
        'Save As
        longstatus = swDoc.SaveAs3("C:\Sample Files\Drawing1.pdf", 0, 2)


    End Sub
    ''' <summary>
    ''' The SldWorks swApp variable is pre-assigned for you.
    ''' </summary>
    Public swApp As SldWorks
End Class
